addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputConverterXML.java

```diff
@@ -0,0 +1,167 @@
+    private final Map<String, String> inputMap;
+    /**
+     * Holds all the inputs names that should NOT be transformed by the {@link #convertNameToID(String)} method.
+     */
+    private final Set<String> inputsWithoutMapping;
+
+    public InputConverterXML(AbstractConnection con, String inputMapConfig) {
+        this.comReference = new WeakReference<>(con);
+
+        logger.trace("User defined mapping: {}", inputMapConfig);
+        this.inputMap = createMapFromSetting(inputMapConfig);
+
+        try {
+            this.inputsWithoutMapping = createInputsWithoutMapping();
+            logger.trace("These inputs will not be mapped: {}", inputsWithoutMapping);
+        } catch (IOException | ReceivedMessageParseException e) {
+            throw new RuntimeException("Could not communicate with the device", e);
+        }
+    }
+
+    /**
+     * Creates a map from a string representation: "KEY1=VALUE1,KEY2=VALUE2"
+     *
+     * @param setting
+     * @return
+     */
+    private Map<String, String> createMapFromSetting(String setting) {
+        Map<String, String> map = new HashMap<>();
+
+        if (!StringUtils.isEmpty(setting)) {
+            String[] entries = setting.split(","); // will contain KEY=VALUE entires
+            for (String entry : entries) {
+                String[] keyValue = entry.split("="); // split the KEY=VALUE string
+                if (keyValue.length != 2) {
+                    logger.warn("Invalid setting: {} entry: {} - KEY=VALUE format was expected", setting, entry);
+                } else {
+                    String key = keyValue[0];
+                    String value = keyValue[1];
+
+                    if (map.putIfAbsent(key, value) != null) {
+                        logger.warn("Invalid setting: {} entry: {} - key: {} was already provided before", setting, entry, key);
+                    }
+                }
+            }
+        }
+        return map;
+    }
+
+    private Set<String> createInputsWithoutMapping() throws IOException, ReceivedMessageParseException {
+        // Tested on RX-S601D, RX-V479
+        Set<String> inputsWithoutMapping = Stream
+                .of(INPUT_SPOTIFY, INPUT_BLUETOOTH)
+                .collect(toSet());
+
+        Set<String> nativeInputNames = XMLProtocolService.getInputs(comReference.get(), Zone.Main_Zone)
+                .stream()
+                .filter(x -> x.isWritable())
+                .map(x -> x.getParam())
+                .collect(toSet());
+
+        // When native input returned matches any of 'HDMIx', 'AUDIOx' or 'NET RADIO', ensure no conversion happens.
+        // Tested on RX-S601D, RX-V479
+        nativeInputNames.stream()
+                .filter(x -> startsWithAndLength(x, "HDMI", 1)
+                        || startsWithAndLength(x, "AUDIO", 1)
+                        || x.equals(INPUT_NET_RADIO)
+                        || x.equals(INPUT_MUSIC_CAST_LINK))
+                .forEach(x -> inputsWithoutMapping.add(x));
```

Comment:
Looks like almost every input (Hdmi, Audio, Spotify, Bluetooth) is not allowed to be mapped. Or do I misread this code?

Reply:
All right